tools/DevDiagnostics/DevHome.DevDiagnostics/Controls/ThemeAwareWindow.cs

```diff
@@ -0,0 +1,183 @@
+
+    public ThemeAwareWindow()
+    {
+        _dispatcher = DispatcherQueue.GetForCurrentThread();
+
+        // Precreate the brushes for the caption buttons.
+        // In Dark Mode, the active state is white, and the inactive state is translucent white.
+        // In Light Mode, the active state is black, and the inactive state is translucent black.
+        var color = Colors.White;
+        _darkModeActiveCaptionBrush = new SolidColorBrush(color);
+        color.A = 0x66;
+        _darkModeInactiveCaptionBrush = new SolidColorBrush(color);
+
+        color = Colors.Black;
+        _nonDarkModeActiveCaptionBrush = new SolidColorBrush(color);
+        color.A = 0x66;
+        _nonDarkModeInactiveCaptionBrush = new SolidColorBrush(color);
+
+        ExtendsContentIntoTitleBar = true;
+        AppWindow.TitleBar.IconShowOptions = IconShowOptions.HideIconAndSystemMenu;
+
+        Closed += ThemeAwareWindow_Closed;
+        Activated += Window_Activated;
+        _uiSettings.ColorValuesChanged += UiSettings_ColorValuesChanged;
+    }
+
+    // Invoked when the user changes system-wide personalization settings.
+    private void UiSettings_ColorValuesChanged(UISettings sender, object args)
+    {
+        _dispatcher.TryEnqueue(ApplySystemThemeToCaptionButtons);
+    }
+
+    private void ThemeAwareWindow_Closed(object sender, WindowEventArgs args)
+    {
+        Activated -= Window_Activated;
+        _uiSettings.ColorValuesChanged -= UiSettings_ColorValuesChanged;
+    }
+
+    internal void Window_Activated(object sender, WindowActivatedEventArgs args)
+    {
+        // This follows the design guidance of dimming our title bar elements when the window isn't activated.
+        // https://learn.microsoft.com/en-us/windows/apps/develop/title-bar#dim-the-title-bar-when-the-window-is-inactive
+        _currentActivationState = args.WindowActivationState;
+
+        if (CustomTitleBarButtons.Count > 0)
+        {
+            UpdateCustomTitleBarButtonsTextColor();
+        }
+    }
+
+    // Invoked when the user changes theme preference in this app's settings.
+    internal void SetRequestedTheme(ElementTheme theme)
+    {
+        Theme = theme;
+
+        if (Content is FrameworkElement rootElement)
+        {
+            rootElement.RequestedTheme = theme;
+
+            if (theme == ElementTheme.Dark)
+            {
+                SetCaptionButtonColors(Colors.White);
+            }
+            else if (theme == ElementTheme.Light)
+            {
+                SetCaptionButtonColors(Colors.Black);
+            }
+            else
+            {
+                ApplySystemThemeToCaptionButtons();
+            }
+        }
+
+        foreach (var window in RelatedWindows)
+        {
+            window.SetRequestedTheme(theme);
+        }
+    }
+
+    // AppWindow.TitleBar doesn't update caption button colors correctly when changed while app is running.
+    // https://task.ms/44172495
```

Comment:
I followed the task, and it looks like that was fixed in WinAppSDK 1.6, can you re-verify?

Reply:
DevHome is not moving to WinAppSDK 1.6 before v0.18. Filed #3803 to track this.